resources/2024-07-secretstore-feature-spec.md

```diff
@@ -0,0 +1,426 @@
+
+## Desired user experience outcome
+
+[existing feature] As an operator, I can define an `Applications.Core/secretStores` resource and deploy it along with an Environment so that the developers I support can securely leverage secrets I manage on their behalf for use in accessing private Recipes. This is a supported scenario in an existing feature, specifically for private [Bicep](https://docs.radapp.io/guides/recipes/howto-private-bicep-registry/) and [Terraform](https://docs.radapp.io/guides/recipes/terraform/howto-private-registry/) Recipes.
+
+[existing feature] As a developer, I can reference an `Applications.Core/secretStores` in my `Applications.Core/gateways` resource definition so that Radius can set up the gateway with the TLS certificate required for HTTPS connections. This is a supported scenario in an existing feature, see [here](https://docs.radapp.io/guides/author-apps/networking/tls/#step-3-add-a-gateway).
+
+[existing feature] As a developer, I can reference an `Applications.Core/secretStores` in my `Applications.Core/containers` resource definition so that Radius will inject secrets as environment variables into my container at deploy time so that credentials can be provided to the container for authentication, etc. This is a supported scenario in an existing feature, see [here](https://docs.radapp.io/reference/resource-schema/core-schema/container-schema/#container).
+
+[proposed feature] As a developer, I can reference an `Applications.Core/secretStores` in my `Applications.containers` resource definition under the `volumes` object so that Radius will write the secret into a file on disk/memory and mount that volume into the container at deploy time. I no longer have to store secrets as plain text in the `properties.secrets` field of my resource for authentication, etc.
+
+[proposed feature] As a developer, I can reference an `Applications.Core/secretStores` in my `Applications.Datastores/*`, `Applications.Messaging/*`, or `Applications.Extenders` resource definition so that Radius will use the password internally to authenticate into the resource at deploy time. I no longer have to store secrets as plain text in the `properties.secrets` field of my resource for authentication, etc.
+
+[proposed feature] As a developer, I can reference an `Applications.Core/secretStores` in my `Applications.Dapr/*` resource definition so that Radius will use the Radius secret store to manage secrets for use in my Dapr components. I no longer have to create a Dapr secret store and manage the secrets in the Dapr secret store to be able to reference it in my `Applications.Dapr/*` resource.
+
+[future feature] As a developer, I can specify a `Applications.Core/secretStores` in my custom user-defined resource once the [user-defined types](https://github.com/radius-project/design-notes/blob/main/architecture/2024-07-user-defined-types.md) feature is implemented.
```

Comment:
Should we create feature requests in github issues/on the roadmap just to track them, might also help get community inputs if someone tries to look up these features?